### PR TITLE
[Feat]: Added VideoListing page

### PR DESCRIPTION
**Pull request by parthpandyappp**
- Added VideoListing page
- Added VideoDetail page
- Added Sidebar

Currently, the two buttons `Watch Later` & `Add to Playlist` aren't functioning. This PR is only meant for listing the video data.

### CODE REVIEW

File: src/pages/Explore.jsx [hunk 1]
```diff
@@ -1,5 +1,40 @@
+import { useVideoData } from "../custom-hooks";
+import { VideoCard } from "../components";
+
 function Explore() {
-  return <h1>Explore</h1>;
+  const videos = useVideoData();
+  console.log(videos);
```
Comment (sonishreyas) [inline src/pages/Explore.jsx]: you can remove console log from code

**Review by sonishreyas — APPROVED** [review]
COde looks good and easy to understand

**Comment by parthpandyappp** [comment]
Thanks @sonishreyas for the PR review :+1:

File: src/components/VideoCard.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  return (
+    <Link to={`/video/${video._id}`}>
+      <div className="card-main">
+        <img className="img-preview" src={video.imgPreview} alt="vid" />
+
+        <div className="vid-info">
+          <h3 className="vid-title">{video.title}</h3>
+          <p className="vid-desc">{video.description}</p>
+          <p className="vid-desc">Created by: {video.creator}</p>
+        </div>
```
Comment (Sharmaryan) [inline src/components/VideoCard.jsx:13]: I think you can destructure all properties from video object

File: src/custom-hooks/videoData.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  const fetchData = async () => {
+    const getRequest = await axios.get("/api/videos");
+    setData(getRequest.data.videos);
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
```
Comment (Sharmaryan) [inline src/custom-hooks/videoData.js:15]: you can implement Immediately Invoked Function Expression(iife) here

File: src/styles/videolist.css [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+}
+.side-nav p:nth-child(even) {
+  background-color: var(--primary-color);
+  border-radius: 0.5rem;
+  color: var(--slate-white);
+}
+.side-nav p:nth-child(odd) {
+  background-color: #fab2be;
+  border-radius: 0.5rem;
+  /* color: var(--slate-white); */
```
Comment (Sharmaryan) [inline src/styles/videolist.css]: remove this one

File: src/styles/videolist.css [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  padding-left: 0.5rem;
+}
+
+.utils-opts {
+  display: flex;
+  justify-content: center;
+
+  gap: 0.5rem;
+  padding: 0.5rem;
+}
```
Comment (Sharmaryan) [inline src/styles/videolist.css:89]: for values less than 1rem we can go with px

**Review by Sharmaryan — APPROVED** [review]
suggest few changes, also in your sidebar you can set cursor to pointer.

**Comment by parthpandyappp** [comment]
Thanks Aryan for the PR review :+1:

**Pull request merged**